program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,88 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+import logging
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+from program_intent_engagement.apps.core.models import ProgramIntent, User
+
+log = logging.getLogger(__name__)
+
+
+class ProgramIntentsView(APIView):
+    # todo: add expected program intent data
+    """
+    View to insert program intents.
+
+    Given a program intent, this view will either insert the program intent. If the given program intent already
+    exists, no action is taken (no duplicate row).
+
+    Path: /api/[version]/program_intents
+
+    Accepts: [POST]
+
+    HTTP POST
+    Inserts a Program Intent.
+    Expected POST data:
+    POST data parameters:
+     * program_uuid: The uuid of the program intent.
+     * reason: The reason for how intent was measured.
+     * certainty: The certainty of the intent for the user.
+     * effective_timestamp: When the event occurred that created the intent.
+    Returns:
+     * 200: The program intent was inserted.
+     * 400: The program intent request was invalid.
+     * 409: Duplicate program intent was not inserted.
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    # todo: do I need to add these? look into what these do
+    # serializer_class = ProgramIntentSerializer
+    # queryset = ProgramIntent.objects.all
+    # lookup_field = "user__lms_user_id"
```

Comment:
You will have to add your `ProgramIntentUserPermissions` here through the `permission_classes` attribute in order for it to run as part of your view.

Reply:
Thanks for clarifying @MichaelRoytman ! Resolving per our convo! :)